### DISCUSSION TITLE
Idea: on the fly subsetting for websites

### DISCUSSION THREAD

**Discussion by ollimeier** [googlefonts/fontations#658] [Ideas]
While using fontations an idea popped up into my head – and I am almost sure you have thought about it already.
Wouldn't it be perfect if browsers would use fontations by default and read the character used by a homepage and create a font subset based on that? I guess that would improve loading dramatically for large fonts like CJK or Color fonts.

This idea is not new, I know (keywords dynamic subsetting), but it feel very possible with Rust and fontations.

Any thoughts on that?

**Comment by cmyr** [comment]
This functionality is already something that large web font providers implement server side (it needs to be server side because the main motivation is to reduce the size of the font file sent to the client) and this is functionality that we expect fontations to implement eventually. :)

**Reply by garretrieger** [reply]
Also there's an upcoming W3C standard (incremental font transfer: https://w3c.github.io/IFT/Overview.html) that could make this available natively in the browser.

**Reply by yisibl** [reply]
@garretrieger Does IFT have any plans for this year? Can it be officially released? I'm really looking forward to this feature.

**Reply by garretrieger** [reply]
We ended up shifting directions with how the IFT mechanism works. The plan is to now embed a map from (codepoints, features, design space) to patches that add it. Notably this now provides a path for an IFT font that is fully statically hosted (but still could be dynamic). You can see a demo of the new approach here: https://garretrieger.github.io/ift-demo/

As a result of the shift we're currently in the process of rewriting the specification, but getting close to having the rewrites finished. https://w3c.github.io/IFT/Overview.html has the current draft of the new approach.
